vmdb/app/models/service_template_orchestration.rb

```diff
@@ -0,0 +1,54 @@
+class ServiceTemplateOrchestration < ServiceTemplate
+  def create_subtasks(_parent_service_task, _parent_service)
+    # no sub task is needed for this service
+    []
+  end
+
+  def deploy_orchestration_stack(options)
+    params = {}
+    options.each do |attr, val|
+      if attr.start_with? 'dialog_param_'
+        params[attr['dialog_param_'.size..-1]] = val
+      elsif attr.start_with? 'password::dialog_param_'
+        params[attr['password::dialog_param_'.size..-1]] = MiqPassword.decrypt(val)
+      end
+    end
+
+    stack_name = options['dialog_stack_name']
+    on_failure = options['dialog_stack_onfailure']
+    timeout = options['dialog_stack_timeout']
+    stack_options = {:parameters => params, :disable_rollback => on_failure != 'ROLLBACK'}
+    stack_options[:timeout] = timeout.to_i if timeout && !timeout.blank?
+
+    ems = ems_provider(options)
+    template = orchestration_template(options)
+
+    begin
+      ems.stack_create(stack_name, template, stack_options)  # return nil if no exception
+    rescue MiqException::MiqOrchestrationProvisionError => err
+      err.message
+    end
+  end
+
+  def orchestration_stack_status(options)
+    ems = ems_provider(options)
+    begin
+      ems.stack_status(options['dialog_stack_name'])
+    rescue MiqException::MiqOrchestrationStatusError => err
+      # naming convention requires status to end with "failed"
+      return "check_status_failed", err.message
+    end
+  end
+
+  private
+
+  def ems_provider(options)
+    ems_id = options['dialog_stack_provider']
+    ExtManagementSystem.find(ems_id)
+  end
+
+  def orchestration_template(options)
+    template_id = options['dialog_stack_template']
+    OrchestrationTemplate.find(template_id)
+  end
```

Comment:
No. Options are user provided through dialog. They cannot be persisted in the service template.